src/screensComponents/modal/index.js

```diff
@@ -37,6 +36,24 @@ export default class Modal extends BaseComponent {
     overlayBackgroundColor: PropTypes.string
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      visible: props.visible
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const {visible, overlayBackgroundColor} = nextProps;
+    if (Constants.isAndroid && !!overlayBackgroundColor && prevState.visible !== visible) {
+      const statusBarColor = visible ? Colors.getOpaqueColor(overlayBackgroundColor, 0.8) : ThemeManager.primaryColor;
```

Comment:
I prefer to add a new prop since we might be able to add the theme logic for the color. 
by using `ThemeManager.primaryColor` we're always going to have a static color which is incorrect..